[Li Yuxuan] iP

CarrieLi1015 opened the pull request:
### Duke

> “Nothing is particularly hard when you divide it into small jobs.” – Henry Ford ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

+ *text-based* :laughing:
+ *easy to learn* :wink:
+ ~~FAST~~ ***SUPER FAST*** to use :star_struck:

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you.

And it is **FREE**! 

Features: 
- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```js
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

CarrieLi1015 committed:
Week 2_level-1

CarrieLi1015 committed:
Week 2_level-2

CarrieLi1015 committed:
Week 2_level-3

CarrieLi1015 committed:
Week 2_level-4

CarrieLi1015 committed:
Week 2_TextUiTesting

CarrieLi1015 committed:
Week 2_level-5

CarrieLi1015 committed:
Week 2_level-6

CarrieLi1015 committed:
Week 3_level-7

CarrieLi1015 committed:
Week 3_level-8

CarrieLi1015 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Week 3_level-7

CarrieLi1015 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Week 3_level-8

# Conflicts:
#	src/main/java/Duke.java

Resolution: keep the latest version

CarrieLi1015 committed:
Week 3_A-MoreOOP

CarrieLi1015 committed:
Week 3_A-Packages

CarrieLi1015 committed:
Merge branch 'add-gradle-support' of https://github.com/CarrieLi1015/ip

CarrieLi1015 committed:
Week 3_A-JUnit

CarrieLi1015 committed:
Week 3_A-Jar

CarrieLi1015 committed:
Week 3_A-JavaDoc

CarrieLi1015 committed:
Week 3_A-CodingStandard

CarrieLi1015 committed:
Week 3_branch-Level-9

CarrieLi1015 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Week 3_A-JavaDoc

CarrieLi1015 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Week 3_A-CodingStandard

# Conflicts:
#	src/main/java/Duke.java

Resolution: keep the latest version

CarrieLi1015 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Week 3_branch-Level-9

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

Resolution: keep the latest version

CarrieLi1015 committed:
Week 3_Complete modify TaskList.java

CarrieLi1015 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Week 3_Complete modify TaskList.java

CarrieLi1015 committed:
Week 3_Modify Parse.java

CarrieLi1015 committed:
Week3_Additional edit for Parser.java

CarrieLi1015 committed:
Add checkstyle.xml file and suppression.xml file for style checking

Using checkstyle.xml to check if the codes follow the required
rules. Several changes are made to cater the rules.

minosx31 commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,52 @@
+ * Different exceptions correspond to different message.
+ */
+public class DukeException extends Exception {
+    private final String eMessage;
```
Perhaps could change `eMessage` to `errorMessage` to improve readability.

CarrieLi1015 replied:
Noted! Thank you for your advance! :)

minosx31 reviewed (commented):
Besides one of the variable names that can be improved, everything else looks good to me. Nice work! 👍🏼

CarrieLi1015 committed:
Implement GUI using JavaFX

Instead of using Previous UI class for frontend,
now using GUI for the interaction between users and
Duke.

Add in two fxml files and modify Duke.java,
Parser.java, and TaskList.java that cater to
output in the GUI window

CarrieLi1015 committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Implement GUI using JavaFX

anthonynmh commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,195 @@
+                }
+            }
+        }
+        if (!ifDeadlineExist) {
```
Should `ifDeadlineExist` be renamed to `hasDeadline` to improve readability?

anthonynmh reviewed (commented):
LGTM. Code mostly follows the coding standards already 😄 !

CarrieLi1015 committed:
Add assertions

Parser class: assertions for two switch cases,
mainly target the default, mark, unmark, and delete.

TaskList class: assertions for mark, unmark and delete method.

Add assertions to prevent the Duke from crushing.

Verfiying the assumptions mainly about invariants,
assumn the user inputs are always within certain range.

SHni99 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,131 @@
+        String recordDescription = userInput.substring(7);
+        Task item = null;
+
+        switch (recordType) {
```
Suggest to put these chunks of code into methods inside the execute function in diff task class (etc, todo, event, deadline)

SHni99 reviewed (commented):
Just some tidying of code and it is good to go

CarrieLi1015 committed:
Code quality check and refactor

Make sure every method is within 30 lines long. Eliminiate
deep nested and try to implement the SLAP principle and
happy path prominent.

Parser class: further clean up the code. Now the parse input only
call functions.

TaskList class: all the task handling is happen inside task list.

Code quality check is done to make the code neater and
easier to read and understand.

CarrieLi1015 committed:
Merge pull request #1 from CarrieLi1015/branch-A-Assertions

Add assertions

CarrieLi1015 committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assertions

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Main.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java

Resolve the conflict by merging the changes in branch-A-Assertions
and branch-A-CodeQuality.

CarrieLi1015 committed:
Merge pull request #2 from CarrieLi1015/branch-A-CodeQuality

Code quality check and refactor

CarrieLi1015 committed:
Add C-Sort extension

Allow user to sort the deadline or event (by the from date)
in ascending chronological order.

It is more convenience for user if can list down the date
in ascending order, so that they can prioritise their tasks.

CarrieLi1015 committed:
Merge branch 'branch-C-Sort'

* branch-C-Sort:
  Add C-Sort extension

CarrieLi1015 committed:
Rearrange the package

Move Duke.java, DialogBox.java, Launcher.java, Main.java, MainWindow.java
under duke package.

It is easier for other to refer and follow the coding standard.

CarrieLi1015 committed:
Update UI

The window has changed to be responsive, the duke's and
user's response is differentiate by using different colour
background. Add padding between image and text label.

Update UI to make it more user friendly.

CarrieLi1015 committed:
Add more error handling

Improve the code to handle all the error I can anticipate. Mainly
focus on parse the user input and make sure duke can response
correctly even the user input is in the wrong format.

Improve user experience.

CarrieLi1015 committed:
Create user guide

A detailed user guide can be found under doc folder.

Convenient for user to follow.

CarrieLi1015 committed:
Edit README

Remove the unnecessary blank space below.

CarrieLi1015 committed:
Update README.md

CarrieLi1015 committed:
Update README.md

CarrieLi1015 committed:
Update README.md

CarrieLi1015 committed:
Update README.md

CarrieLi1015 committed:
Update README.md
